File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    static int[] chosen;
    static List<int[]> list;
    static HashMap<Integer,Integer> map;
    
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int sum = 0;
        map = new HashMap<>();
        for (int i : tangerine) {
            map.put(i, map.getOrDefault(i,0) + 1);
        }
        List<Integer> keyList = new ArrayList<>(map.keySet());
        keyList.sort((k1,k2) -> map.get(k2) - map.get(k1));
        
        for (Integer n : keyList) {
            if (k <= 0) {
                break;
            }

            answer++;
            k -= map.get(n);
        }
        return answer;
    }
    
    
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    arr = [0] * 10000001
    for i in range(len(tangerine)) :
        arr[tangerine[i]] += 1
    arr.sort(reverse=True)
    
    cnt = 0
    for i in range(len(arr)) :
        if cnt >= k :
            break
        cnt += arr[i]
        answer += 1
    return answer